pymodbus/server/async_io.py

```diff
@@ -180,19 +180,23 @@ def execute(self, request, *addr):
         if self.server.request_tracer:
             self.server.request_tracer(request, *addr)
 
-        broadcast = False
+        if self.server.broadcast_enable and not request.slave_id:
```

Comment:
Moving this out from the try/except, will probably cause problems with the remote datastore.

Reply:
case broadcast before:
1. a response is created for the last slave_context
2. a NoSuchSlaveException could not happen, because of the iterator
3. a ServerContext, SlaveContext, datastore could rise an exception within the loop
4. data integrity is corrupted, because only slave contexts till the exception were updated
5. a misleading log messages and a response is created
6. the created responses are never send

case broadcast pull request:
1. responses are ignored
2. raised exceptions in ServerContext, SlaveContext, datastore within the loop bubble up
3. server will probably terminate or restart

I think exceptions should be handled in the server context.
Note the misleading error messages is still there for non broadcast use cases